lib/node_modules/@stdlib/math/base/special/acoth/lib/acoth.js

```diff
@@ -0,0 +1,65 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var atanh = require( '@stdlib/math/base/special/atanh' );
+
+
+// MAIN //
+
+/**
+* Computes the inverse hyperbolic cotangent of a number.
+*
+* @param {number} x - input value
+* @returns {number} inverse hyperbolic cotangent (in radians)
+*
+* @example
+* var v = acoth( 0.0 );
+* // returns 0.0
+*
+* @example
+* var v = acoth( 0.5 );
+* // returns NaN
+*
+* @example
+* var v = acoth( 1.0 );
+* // returns Infinity
+*
+* @example
+* var v = acoth( 2.0 );
+* // returns ~0.5493
+*
+* @example
+* var v = acoth( NaN );
+* // returns NaN
+*/
+function acoth( x ) {
+	if (x === 0.0) {
```

Comment:
Why is this special case present? The inverse hyperbolic cotangent is not defined for `0.0`.

Reply:
atanh should return 0.0 but is returning NaN. I probably missed something, but I could not figure it out why.

Reply:
`atanh` **does** return `0.0`. To see this in action, start the stdlib REPL,

```
$ make repl
```

In the REPL,

```
> base.atanh( 0.0 )
0.0
```

For the inverse hyperbolic cotangent `acoth`, the equivalent `atanh` call is

```
> base.atanh( 1.0 / 0.0 )
NaN
```

This result is the same in Julia, Python, etc.

That the result should be `NaN` and not `0.0` can be confirmed by examining the plot of `acoth` where the domain is not defined in the region between (-1,1), as also stated in this package's documentation.

Hope this helps! :)